pyteal/ast/expr.py

```diff
@@ -8,10 +8,30 @@
 if TYPE_CHECKING:
     from ..compiler import CompileOptions
 
+import inspect
+
+
+def depth(init):
+    def wrapper(*args, **kwargs):
+        if not hasattr(args[0], "depth"):
+            args[0].depth = 0
+        args[0].depth += 1
+
+        if not hasattr(args[0], "frame"):
+            args[0].frames = inspect.stack()
+            args[0].kframe = args[0].frames[args[0].depth * 2 - 1]
+
+        return init(*args, **kwargs)
+
+    return wrapper
+
 
 class Expr(ABC):
     """Abstract base class for PyTeal expressions."""
 
+    def __init_subclass__(cls):
```

Comment:
Turns out, I don't need all this heavy machinery... will trim this down to its bare essentials